bundles/org.openhab.binding.bluetooth.am43/src/main/java/org/openhab/binding/bluetooth/am43/internal/AM43Handler.java

```diff
@@ -0,0 +1,474 @@
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.core.util.HexUtils;
+import org.openhab.binding.bluetooth.BluetoothCharacteristic;
+import org.openhab.binding.bluetooth.BluetoothCompletionStatus;
+import org.openhab.binding.bluetooth.BluetoothDevice.ConnectionState;
+import org.openhab.binding.bluetooth.ConnectedBluetoothHandler;
+import org.openhab.binding.bluetooth.am43.internal.command.AM43Command;
+import org.openhab.binding.bluetooth.am43.internal.command.ControlCommand;
+import org.openhab.binding.bluetooth.am43.internal.command.GetAllCommand;
+import org.openhab.binding.bluetooth.am43.internal.command.GetBatteryLevelCommand;
+import org.openhab.binding.bluetooth.am43.internal.command.GetLightLevelCommand;
+import org.openhab.binding.bluetooth.am43.internal.command.GetPositionCommand;
+import org.openhab.binding.bluetooth.am43.internal.command.GetSpeedCommand;
+import org.openhab.binding.bluetooth.am43.internal.command.ResponseListener;
+import org.openhab.binding.bluetooth.am43.internal.command.SetPositionCommand;
+import org.openhab.binding.bluetooth.am43.internal.command.SetSettingsCommand;
+import org.openhab.binding.bluetooth.am43.internal.data.ControlAction;
+import org.openhab.binding.bluetooth.am43.internal.data.Direction;
+import org.openhab.binding.bluetooth.am43.internal.data.MotorSettings;
+import org.openhab.binding.bluetooth.am43.internal.data.OperationMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tec.uom.se.unit.MetricPrefix;
+
+/**
+ * The {@link AM43Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Connor Petty - Initial contribution
+ */
+@NonNullByDefault
+public class AM43Handler extends ConnectedBluetoothHandler implements ResponseListener {
+
+    private final Logger logger = LoggerFactory.getLogger(AM43Handler.class);
+
+    private @Nullable AM43Configuration config;
+
+    private volatile @Nullable AM43Command currentCommand = null;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private @Nullable ExecutorService commandExecutor;
+
+    private @Nullable MotorSettings motorSettings = null;
+
+    public AM43Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
```

Comment:
This method is deprecated. Please set the thing status to UNKNOWN here instead.

Reply:
The parent handler class is ConnectedBluetoothHandler, so the super.initialize() is required.

Reply:
Outch, you've got me there :-) You're obviously right, I missed that.

Reply:
I don't blame you, the bluetooth binding is a unique case 😄